### PR TITLE
Error handling, Stability improvements for RunFromNetwork

### PR DESCRIPTION
**Pull request by tilk48**
- **Error handling**
The RunFromNetwork widget now accepts a callback for when an error occurs during the loading or build process. This callback receives the error object and return a widget to be rendered on screen.

- **Stability improvements**

1. You can now actively disable or enable debug mode for the widget as the use of the localhost url introduced unpredictable instabilities.
2. The loading does not crash anymore when the widget is in debug mode but no symbol file could be found on the server

### CODE REVIEW

File: lib/runFromNetwork.dart [hunk 1]
```diff
@@ -152,20 +179,32 @@ class _RunFromNetwork extends State<RunFromNetwork>
     }
 
     maybeReload();
-    timer = Timer.periodic(
+    /*timer = Timer.periodic(
```
Comment (chgibb) [inline lib/runFromNetwork.dart:182]: Hot reload is driven by polling. If the timer isn't initialized, then the content will never reload. Not initializing the timer is also causing a crash trying to call `timer.cancel` in `dispose`.

**Comment by chgibb** [comment]
Thank you so much @tilk48 for contributing! I've taken a look and left a comment.

File: lib/runFromNetwork.dart (hunk 1 above)
Comment (tilk48) [reply lib/runFromNetwork.dart]: You are absolutely right, I just forgot to merge my branches correctly. Should be fixed in the new commit

**Comment by tilk48** [comment]
- Added an explicit debug flag for developers who want to use that mode

- Uncommented the timer for hot reload

File: lib/runFromNetwork.dart [hunk 2]
```diff
@@ -40,7 +44,9 @@ class RunFromNetwork extends StatefulWidget {
     this.downloadHash,
     this.downloadByteCodeImage,
     this.downloadDebugInfo,
+    this.errorBuilder,
     this.customNamespaces,
+    this.debugMode = false,
```
Comment (chgibb) [inline lib/runFromNetwork.dart:49]: I just noticed this is failing out the debug mode tests. Should this instead be default initialized to `kDebugMode` if the user doesn't specify?

Comment (tilk48) [reply lib/runFromNetwork.dart]: Thats a very good idea, I added that in the new commit

**Pull request merged**